### PR TITLE
Added New tag on Activity

### LINKED ISSUE #7353: Introducing "new" tag for new activity cards
**Please describe the desired behavior.**
Describe the solution you'd like.
It will be good if we put a new tag in user's activity which are added after the user's last login.
**Current Behaviour**
Screenshot 2020-01-26 at 4 59 50 PM

**Expected Behaviour**
Screenshot 2020-01-26 at 4 54 10 PM

**Additional context (optional)**
Feel free to include any applicable: code, screenshots, or gifs.

### PR DESCRIPTION
**Pull request by sssash18**
Fixes #7353  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
**New View**
Screenshot 2020-04-23 at 11 32 00 AM

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/views/dashboard/_node_wiki.html.erb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 <div class="col-lg-6 note-container-wiki">
   <div class="note note-pane note-wiki" data-index="<%= index %>">
-
+    
+    <% if current_user && node.is_new_to_user(node, current_user) %>
```
Comment (gauravano) [inline app/views/dashboard/_node_wiki.html.erb:4]: The usage here will raise the error as node in this scenario is an object of revision. 

Add delegation in revision to use the method from node https://simonecarletti.com/blog/2009/12/inside-ruby-on-rails-delegate/


Also, you don't have to paas node as parameter when you're calling using the related object of it.

File: app/views/dashboard/_node_question.html.erb [hunk 2]
```diff
@@ -4,6 +4,12 @@
     <div class="header-icon"><i class="fa fa-question-circle"></i> <a href="/q/<%= node.id %>"><i class="fa fa-link"></i></a></div>
 
     <div style="overflow: hidden;">
+      
+      <% if current_user && node.is_new_to_user(node, current_user) %>
```
Comment (gauravano) [inline app/views/dashboard/_node_question.html.erb:8]: Please see implement change here also as asked for `_node_wiki.html.erb`

File: app/views/dashboard/_node_comment.html.erb [hunk 3]
```diff
@@ -11,6 +11,12 @@
       <a href="/n/<%= node.parent.id %>"><i class="fa fa-link"></i></a>
     </div>
 
+    <% if current_user && (node.created_at.to_f > current_user.last_login_at.to_f) %>
```
Comment (gauravano) [inline app/views/dashboard/_node_comment.html.erb:14]: implement DRY. use the model method.

File: app/models/node.rb [hunk 4]
```diff
@@ -1096,4 +1096,8 @@ def notify_callout_users
       NodeMailer.notify_callout(self, user).deliver_now if user.username != author.username
     end
   end
+
+  def is_new_to_user(node, user)
```
Comment (gauravano) [inline app/models/node.rb:1100]: remove node from parameter list and rectify the definition

Comment (gauravano) [reply app/models/node.rb:1100]: Also, please write test for this method

**Comment by gauravano** [comment]
**Suggestion about naming branches:** While naming branches which I also learned sometime back:
1. Every branch can be grouped in one of the following buckets: `feature/`, `refactor/`, `bugfix/` , `hotfix/`
2. Give a very brief but descriptive enough name to the branch like `feature/new-tag-post` rather than naming branch wrt it's respective feature directly

Good branch names are as important and helpful as good commit messages in situations we have to do a quick revert/hotfix.

Also on a side note, as a convention commit messages should be in the present tense (what is the implication of this particular commit), example: adds check for the node data creation or checks node if eligible for new tag

**Comment by sssash18** [comment]
@gauravano Thanks for your detailed review! I will implement the changes soon...

**Comment by jywarren** [comment]
Hi, it's been a little while, but I wonder if you're still available to look over @gauravano's requests? Thank you and if not, perhaps we can open this up to someone else? Thanks, everyone!

**Pull request closed without merging**